waku/v2/waku_core/topics/sharding.nim

```diff
@@ -0,0 +1,113 @@
+## Waku autosharding utils
+##
+## See 51/WAKU2-RELAY-SHARDING RFC: https://rfc.vac.dev/spec/51/#automatic-sharding
+
+when (NimMajor, NimMinor) < (1, 4):
+  {.push raises: [Defect].}
+else:
+  {.push raises: [].}
+
+import
+  nimcrypto,
+  std/math,
+  std/sequtils,
+  std/algorithm,
+  std/strutils,
+  stew/endians2,
+  stew/results,
+  stew/byteutils
+
+import
+  ./content_topic,
+  ./pubsub_topic
+
+const ClusterIndex* = 49152
```

Comment:
Hmm...my understanding is that TWN actually is a network - it is one instantiation of an opinionated relay network. You could then establish a multi-level forking terminilogy as in "soft fork" where you use the same discovery as TWN, but different shards (i.e. nodes participating in TWN may not relay your messages), or a "hard fork" where you also split the discovery layer (i.e. create a completely separate network somehow). My point of view could be over-simplified though:)

Anyway, +1 for simplifying the cluster/namespace numbering

Reply:
Not that set on specific name here, but to me "network" does not necessarily have to imply logical centralization - it's just a collective noun for the "suggested" cluster and index range for default WN routing. That said, I'm happy if we continue calling it "cluster", though we may want to think about whether we have a "main" cluster and a "test" cluster of shards and to rather opt for simpler cluster indexes (e.g. `1` vs `49152`).

Reply:
In our space, I think when ppl hear "mainnet" they would go :bulb: blockchain but maybe that's just me.

Your also right that it IS a network but because of the space we live in words have cultural weight to them.
We may want to find different words to differentiate ourselves.